react-validated-input-component/src/validated-input.jsx

```diff
@@ -0,0 +1,75 @@
+import React, { useState } from 'react';
+
+export default function ValidatedInput() {
+  const [password, setPassword] = useState('');
+  const [errorMessage, setErrorMessage] = useState('A password is required');
+  const [icon, setIcon] = useState('fa-solid fa-xmark');
+
+  const handlePasswordChange = (e) => {
+    const noError = '';
+    const regexSpecialChars = /[!@#$%^&*()?]/g;
+    const regexCapitalChars = /[A-Z]/g;
+    const regexDigits = /[0-9]/g;
+
+    setPassword(e.target.value);
+
+    switch (true) {
+      case (password.length === 0):
+        setErrorMessage('A password is required');
+        setIcon('fa-solid fa-xmark');
+        e.preventDefault();
+        break;
+
+      case (password.length < 7):
+        setErrorMessage('Your password is too short.');
+        setIcon('fa-solid fa-xmark');
+        e.preventDefault();
+        break;
+
+      case !(regexSpecialChars.test(e.target.value)):
+        setErrorMessage('Password requires at least one special character.');
+        setIcon('fa-solid fa-xmark');
+        e.preventDefault();
+        break;
+
+      case !(regexCapitalChars.test(e.target.value)):
+        setErrorMessage('Password requires at least 1 capital letter.');
+        setIcon('fa-solid fa-xmark');
+        e.preventDefault();
+        break;
+
+      case !(regexDigits.test(e.target.value)):
+        setErrorMessage('Password needs at least 1 digit.');
+        setIcon('fa-solid fa-xmark');
+        e.preventDefault();
+        break;
+
+      default:
+        setErrorMessage(noError);
+        setIcon('fa-solid fa-check');
+    }
```

Comment:
I am guessing that you made `errorMessage` and `icon` state because you didn't know how to handle getting them both from this function. Or you weren't sure what to do since you wanted to do the computations in response to a changed password.

The best way to handle this is to have your `onChange` handler only update the state, and then in the body of the component function compute the `errorMessage` and `icon` variables. It's best to keep the handlers simple and single-focused and to have the main component code deal with the component logic.